src/fork_gc.c

```diff
@@ -0,0 +1,757 @@
+#include "fork_gc.h"
+#include "util/arr.h"
+#include "search_ctx.h"
+#include "inverted_index.h"
+#include "redis_index.h"
+#include "numeric_index.h"
+#include "tag_index.h"
+#include "tests/time_sample.h"
+#include <stdlib.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <wait.h>
+
+static void ForkGc_updateStats(RedisSearchCtx *sctx, ForkGCCtx *gc, size_t recordsRemoved,
+                               size_t bytesCollected) {
+  sctx->spec->stats.numRecords -= recordsRemoved;
+  sctx->spec->stats.invertedSize -= bytesCollected;
+  gc->stats.totalCollected += bytesCollected;
+}
+
+static void ForkGc_FDWriteLongLong(int fd, long long val) {
+  write(fd, &val, sizeof(long long));
+}
+
+static void ForkGc_FDWriteBuffer(int fd, char *buff, size_t len) {
+  ForkGc_FDWriteLongLong(fd, len);
+  if (len > 0) {
+    write(fd, buff, len);
+  }
+}
+
+static long long ForkGc_FDReadLongLong(int fd) {
+  long long ret;
+  ssize_t sizeRead = read(fd, &ret, sizeof(long long));
+  if (sizeRead != sizeof(long long)) {
+    return 0;
+  }
+  return ret;
+}
+
+static char *ForkGc_FDReadBuffer(int fd, size_t *len) {
+  *len = ForkGc_FDReadLongLong(fd);
+  if (*len == 0) {
+    return NULL;
+  }
+  char *buff = rm_malloc(*len * sizeof(char *));
```

Comment:
rm_malloc(*len). This line causes you to allocate 8x the amount of memory we actually need.